.github/workflows/build-and-test.yml

```diff
@@ -89,6 +89,47 @@ jobs:
         if: ${{ env.IS_NOOP == 'true' }}
         run: echo "No checks on main or release branches, skipping this job."
 
+  e2e-test:
+    name: Build and e2e-test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ "21" ]
+    env:
+      IS_NOOP: ${{ github.ref_name == 'main' || startsWith(github.ref_name, 'release-') }}
+    steps:
+      - name: Inform on no-op
+        run: echo "Running this as a no-op job... ${{ env.IS_NOOP}}"
```

Comment:
**issue (non-blocking)**: The `env.IS_NOOP` workaround is really a pain to read and understand

I understand this is currently required, because on main we do not (want?) to have a proper build. Maybe it is time to think about review this setup in a followup?

Reply:
sure, if you can think of a better way, I'm all ears!